registry.go

```diff
@@ -40,6 +40,12 @@ type Namespace interface {
 	// which were filled.  'last' contains an offset in the catalog, and 'err' will be
 	// set to io.EOF if there are no more entries to obtain.
 	Repositories(ctx context.Context, repos []string, last string) (n int, err error)
+
```

Comment:
I would be in favor of splitting out these functions into a separate interface and type asserting to use. This would keep from needing to add unsupported and panics everywhere that this is not supported.